analyses/molecular-subtyping-neurocytoma/01-neurocytoma-subtyping.Rmd

```diff
@@ -0,0 +1,77 @@
+---
+title: "Subtype Neurocytoma tumors as central or extra-ventricular"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+author: Krutika Gaonkar for D3b
+date: 2020
+---
+
+Subtyping conditions as per issue [#805](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/805) are as follows :
+
+ - If primary_site == "Ventricles", then molecular_subtype == CNC (central neurocytoma) 
+ - If primary_site != "Ventricles", then molecular_subtype == EVN (extraventricular neurocytoma)
+
+## Set up 
+
+```{r}
+library(tidyverse)
+```
+
+## Input directories and files
+
+```{r}
+data_dir <- file.path("..", "..", "data")
+histologies_df <- read_tsv(file.path(data_dir, "pbta-histologies.tsv"))
+
+```
+
+## Output directory and file
+
+```{r}
+results_dir <- "results"
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+outputfile <- file.path(results_dir,"neurocytoma_subtyping.tsv")
+```
+
+Checking if `pathology_diagnosis` has "Neurocytoma" in their value, and lets view the current short_histology values
+
+```{r}
+neurocytoma_from_pathology <- histologies_df %>%
+  filter(str_detect(pathology_diagnosis, "Neurocytoma")
+                    ) 
+  
+neurocytoma_from_pathology$short_histology
+```
+
+There's a "CNS neuroblastoma" in short_histology let's check if pathology_diagnosis and pathology_free_text_diagnosis agree
+
+```{r}
+neurocytoma_from_pathology %>%
+  filter(short_histology == "CNS neuroblastoma"
+                    ) %>%
+  select(pathology_diagnosis,pathology_free_text_diagnosis)
+```
+
+
+Both pathology_diagnosis and pathology_free_text_diagnosis is Neurocytoma so keeping it in Neurocytoma list
+
+```{r}
+
+neurocytoma_subtype <- neurocytoma_from_pathology %>% 
+  dplyr::mutate(molecular_subtype= 
+                  case_when(primary_site == "Ventricles" ~ "CNC",
+                            primary_site != "Ventricles" ~ "EVN"
+                            )
```

Comment:
Add code comment here about assigning subtypes at this step.

Reply:
Thanks for the review @jharenza ,added comments here!

Reply:
great, thank you!